### ISSUE TITLE
Change the [NotImplemented] attribute to something more descriptive: [ReadOnly]

### ISSUE THREAD

**Issue by rolfbjarne** [xamarin/xamarin-macios#8481]
Sometimes Apple provides an immutable class and a mutable subclass. Example: `AVMetadataItem` and `AVMutableMetadataItem`. If a developer creates an instance of an  `AVMetadataItem` and tries to set a property, a NotImplementedException will be thrown:

```csharp
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
public virtual NSString? MetadataIdentifier {
	[Export ("identifier")]
	get {
            ...
	}
	[NotImplemented ()]
	set {
		throw new NotImplementedException ();
	}
}
```

This is confusing for developers, every now and then we get bug reports about it:

* https://github.com/xamarin/xamarin-macios/issues/8480

Idea: change the generated code to throw something else with a better description:

```csharp
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
public virtual NSString? MetadataIdentifier {
	[Export ("identifier")]
	get {
            ...
	}
	[NotImplemented ()]
	set {
		throw new NotSupportedException ("This member is read-only. Use an instance of the mutable subclass AVMutableMetadataItem instead.");
	}
}
```

**Comment by spouliot** [comment]
To limit the number of _largely_ duplicated strings in the application (impact final size) we should use something like:
```
$"This property is read-only on {typeof (this.GetType ())}. One (or more) subclass(es) might be mutable and allow read/write operations.");
```
In any case the generator does not know the mutable name (even if easy to guess most of the time) bu it knows the current type.

**Comment by rolfbjarne** [comment]
> In any case the generator does not know the mutable name

It could be added to the `[NotImplemented]` attribute.

**Comment by spouliot** [comment]
Yes, if a String.Format is used then it should not duplicate strings in the binary.
It can fall back to a general message (like above) if the argument is missing.

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#14580**
- Please respond to @msftbot[bot].

---

_From @msftbot[bot] on Sunday, April 16, 2023 12:41:54 PM_

Hi @luyikk. We have added the "s/needs-repro" label to this issue, which indicates that we require steps and sample code to reproduce the issue before we can take further action. Please try to create a minimal sample project/solution or code samples which reproduce the issue, ideally as a GitHub repo that we can clone. See more details about creating repros here: https://github.com/dotnet/maui/blob/main/.github/repro.md

This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#14580**
- Please respond to @jfversluis.

---

_From @jfversluis on Sunday, April 16, 2023 12:42:27 PM_

Besides a repro that would come in very handy to diagnose this, could you try to update to .NET 7 and see if that helps things?

**Comment by mattleibow** [comment]
---
**Issue moved from dotnet/maui#14626**
- Please respond to @Kebechet.

---

_From @Kebechet on Thursday, April 20, 2023 6:47:01 AM_

> @Kebechet you need to return the main page with a new window that you've created. So, for your code here
> 
> https://github.com/Kebechet/BlazorHybridBug/blob/main/BlazorHybridApp/BlazorHybridApp/App.xaml.cs#L12
> 
> ```cs
>     protected override Window CreateWindow(IActivationState? activationState)
>     {
>         return new Window(new MainPage());
>     }
> ```

but even though your solution is the right way to go, there is still the problem of inconsistency. Because the only place where it was crashing was deployed ios application in Testflight. It didnt crash in simulator, in local ios device or on other platforms like Android or windows.
